pilot/pkg/serviceregistry/aggregate/controller.go

```diff
@@ -178,7 +178,10 @@ func (c *Controller) Services() ([]*model.Service, error) {
 		}
 
 		if r.Provider() != provider.Kubernetes {
-			services = append(services, svcs...)
+			for _, s := range svcs {
+				services = append(services, s)
+				smap[s.Hostname] = s
```

Comment:
Isn't this changing the previous behavior say if two SEs with same hostname

Reply:
Yes - this is incorrect. Need to update it